test/query.test.ts

```diff
@@ -335,6 +336,23 @@ describe('query', () => {
        assert.equal(str, 'bin(a,q)');
     });
 
+
+    it('should return correct fieldDef string for bin field with maxbins', () => {
+       const str = stringifyEncodingQueryFieldDef({
+         channel: Channel.X, field: 'a', type: Type.QUANTITATIVE, bin: {maxbins: 20}
+       });
+       assert.equal(str, 'bin(a,q,maxbins=20)');
+    });
+
+
+    it('should return correct fieldDef string for scale with scaleType log', () => {
+       const str = stringifyEncodingQueryFieldDef({
+         channel: Channel.X, field: 'a', type: Type.QUANTITATIVE, scale: {type: ScaleType.LOG}
+       });
+       assert.equal(str, 'scale(a,q,scale={type: log})');
```

Comment:
@kanitw 
I have changed it to use bin.maxbins + scale.type. Just to make sure, should the output be this?
bin(a,q,maxbins=20,scale={type: log})